### PR TITLE
Sidebar Tags

### PR DESCRIPTION
**Pull request by jeanlescure**

### CODE REVIEW

**Comment by jeanlescure** [comment]
done

**Comment by indeyets** [comment]
@jeanlescure please try to separate formatting commits and logic-changing commits

File: src/api/controller.js [hunk 1]
```diff
@@ -68,11 +68,31 @@ export default class ApiController {
...
+
+    let q = Post.forge()
+      .query(qb => {
+        qb
+          .where({user_id: req.session.user})
+          .orderBy('posts.created_at', 'desc')
+      });
+
+    let labels = await q.fetchAll({require: false, withRelated: ['labels']})
+      .then(results => results.map(row => row.relations.labels));
```
Comment (indeyets) [inline src/api/controller.js]: either you use `await` either `.then()`

File: src/components/sidebar.js [hunk 2]
```diff
@@ -16,31 +16,54 @@
...
 
+import ApiClient from '../api/client'
+import {API_HOST} from '../config';
+import {getStore, setUserTags, addError} from '../store';
 import CurrentUser from './current-user';
+import TagCloud from './tag-cloud';
+import { defaultSelector } from '../selectors';
 
 export default class Sidebar extends React.Component {
+  static displayName = 'UserPage'
```
Comment (indeyets) [inline src/components/sidebar.js]: Should be `'Sidebar'`

File: src/components/sidebar.js [hunk 3]
```diff
@@ -16,31 +16,54 @@
...
+    if (!this.props.current_user || !this.props.user_tags) {
       return <script/>
     }
 
+    let tagsBlock = (this.props.user_tags.length > 0)? (
+      <div className="layout__row">
+        <h4>Tags</h4>
+        <TagCloud tags={this.props.user_tags}/>
+      </div>
+    ) : (<script/>);
```
Comment (indeyets) [inline src/components/sidebar.js]: let TagCloud do this check (and move `<div>` and `<h4>` into it)

File: src/store.js [hunk 4]
```diff
@@ -392,6 +415,7 @@ function theReducer(state = initialState, action) {
 let initialState = {
   users: {},
   user_posts: {},
+  user_tags: [],
```
Comment (indeyets) [inline src/store.js]: `current_user_tags` maybe?

**Comment by jeanlescure** [comment]
@indeyets changes done

**Comment by indeyets** [comment]
see #15 instead

**Pull request closed without merging**